pwiz_tools/Skyline/Properties/Settings.cs

```diff
@@ -824,6 +824,25 @@ public OptimizationLibraryList OptimizationLibraryList
             }
         }
 
+        [UserScopedSettingAttribute]
+        public SerializableDictionary<string, string> LastArdiaLoginCookieByUsername

```

Comment:
@nickshulman I don't remember why I stored the cookie directly in settings instead of as an Account property that would get stored in the RemoteAccountList, but I'm questioning the wisdom of my decision now. Maybe it had something to do with the immutability of RemoteAccount. What do you think? The cookie is used in lieu of the username/password for the duration of the session and should probably be regenerated every time Skyline is started (or when it stops working).

Reply:
It sounds like this dictionary should just be a static variable somewhere instead of being a property on the Settings object. (If you wanted you could probably just remove the "[UserScopedSettingAttribute]" if you wanted to prevent it from being persisted, but making it a static variable somewhere else is probably cleaner)

Reply:
I don't think there's anything inherently wrong with having transient properties in an object that do not get serialized to XML, but, when you find yourself doing that it's often a good idea to make two separate classes so you don't get confused.
That is, the thing that is stored in the Settings list has a username and password. And then, there's another type of object that has a username, a password and a cookie. And some methods want just the username and password, and other things want a username, a password and a cookie, and the easiest way to indicate which methods are which is which type is to have a different class for these two things.